lib/manager/gitlabci/extract.ts

```diff
@@ -79,3 +82,60 @@ export function extractPackageFile(content: string): PackageFile | null {
   }
   return { deps };
 }
+
+export async function extractAllPackageFiles(
+  config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  const filesToExamine = new Set<string>(packageFiles);
+  const filesAlreadyProcessed = new Set<string>();
+  const results: PackageFile[] = [];
+
+  // extract all includes from the files
+  while (filesToExamine.size > 0) {
+    const file = filesToExamine.values().next().value;
+    filesToExamine.delete(file);
+    filesAlreadyProcessed.add(file);
+
+    const content = await readLocalFile(file, 'utf8');
+    const doc = yaml.safeLoad(content, { json: true }) as any;
+    if (doc?.include && is.array(doc.include)) {
+      for (const includeObj of doc.include) {
+        if (includeObj.local) {
+          const fileObj = (includeObj.local as string).replace(/^\//, '');
+          if (
+            !filesToExamine.has(fileObj) &&
+            !filesAlreadyProcessed.has(fileObj)
+          ) {
```

Comment:
```suggestion
          if (!filesAlreadyProcessed.has(fileObj)) {
```
i think adding again is a noop